### PR TITLE
Limit the length of the error messages

### LINKED ISSUE #71: 2.5 validation range error message is too long
We should make the error message from https://github.com/bcneng/candebot/blob/master/bot/interact.go#L85-L87 shorter

CleanShot 2021-10-01 at 15 28 26@2x

cc @ignasi35 @gonzaloserrano

### PR DESCRIPTION
**Pull request by ignasi35**
Fixes #71

### CODE REVIEW

File: bot/interact_test.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	t.Run("max salary must not be greater than 2.5x the min salary", func(t *testing.T) {
+		_, _, validaterrors := validateSubmission("hamburger", "13", "5")
+		require.Equal(t, "The range MinSalary-MaxSalary is too large. Salary is a relevant field, keep it meaningful to increase offer impact.", validaterrors["max_salary"])
+	})
+
+	t.Run("error messages should be shorter than 150 chars", func(t *testing.T) {
+		_, _, validaterrors := validateSubmission("hamburger", "13", "5")
+		if(len([]rune(validaterrors["max_salary"])) > 150) {
+			t.Errorf("error message is too long: %s", validaterrors["max_salary"])
+		}
```
Comment (ignasi35) [inline bot/interact_test.go:70]: This is the relevant bit of the PR.

File: bot/interact_test.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+)
+
+func requireHasError(t *testing.T, errorMessage string) {
+	require.NotEmpty(t, errorMessage)
+
+	if len([]rune(errorMessage)) > 150 {
+		t.Errorf("error message is too long: %s", errorMessage)
+	}
+
+}
```
Comment (ignasi35) [inline bot/interact_test.go:16]: helper function that validates an error message:

1. is present
2. is shorter than 150 chars (see #71)

**Comment by ignasi35** [comment]
(#70 is merged)
rebased on `master`

File: bot/interact.go [hunk 3]
```diff
@@ -149,7 +149,7 @@ func validateSubmission(messageJobLink, messageMaxSalary, messageMinSalary strin
 		}
 
 		if 2.5*float32(minSalary) < float32(maxSalary) {
-			validationErrors["max_salary"] = "The gap between MinSalary and MaxSalary is rather large. Maybe you should post two different job offers with different responsibilities and required qualifications. Salary is a relevant field, we recommend you try to keep it meaningful to increase the chances of taking the position seriously by potential candidates."
+			validationErrors["max_salary"] = "The range MinSalary-MaxSalary is too large. Salary is a relevant field, keep it meaningful to increase offer impact."
```
Comment (smoya) [inline bot/interact.go:152]: Some suggestions for a more readable version IMHO:
```suggestion
			validationErrors["max_salary"] = "The range min-max salary is too wide. Salary is a relevant field; keep it meaningful to increase offer impact."
```

File: bot/interact_test.go [hunk 4]
```diff
@@ -6,7 +6,17 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func requireHasError(t *testing.T, errorMessage string) {
+	require.NotEmpty(t, errorMessage)
+
+	if len([]rune(errorMessage)) > 150 {
```
Comment (smoya) [inline bot/interact_test.go:12]: You can just do this, right?

```suggestion
	if len(errorMessage) > 150 {
```

Comment (ignasi35) [reply bot/interact_test.go:12]: I just tried it out and it does work as expected. Thanks for the improvement!

**Review by smoya — APPROVED** [review]
LGTM!🎉🎉🎉

**Pull request merged**